### PR TITLE
Bevy 0.11 Update

### LINKED ISSUE #439: Bevy 0.11 Support
I'm starting work on Bevy 0.11 support unless somebody else already has a good start.  I'm just going to follow the change log strictly.

### PR DESCRIPTION
**Pull request by mwbryant**
All examples are tested and work on my machine TM.  

Some concerns I have are with the shader imports, they work but I'm not great with wgsl so I had to duplicate the locations instead of using an import due to the changes in Bevy 0.11.  Also constants were giving an exotic error message so I moved them into the functions that used them.

Also anisotropy_clamp must be set now, I don't personally know the details but it was previously none and wgpu has the default as 1 so I went with that. 

Change detection magic timer matches the recommended time in the migration guide.

Otherwise nothing major, a couple of undocumented fixes I had to find the PRs for and silent failures but nothing too bad.

Closes #439

### CODE REVIEW

File: src/render/shaders/tilemap_vertex.wgsl [hunk 1]
```diff
@@ -1,44 +1,49 @@
-#import bevy_ecs_tilemap::common
+#import bevy_ecs_tilemap::common VertexInput, tilemap_data
```
Comment (rparrett) [inline src/render/shaders/tilemap_vertex.wgsl:1]: We need `tilemap_data` for the `atlas` feature.
```
#import bevy_ecs_tilemap::common tilemap_data
```

**Review by rparrett — CHANGES_REQUESTED** [review]
Thanks for picking this up. I have a couple notes based on a quick skim and an attempt to use the crate with wasm.

I think we should add the feature `webgl2` to the first `dev-dependencies` section for wasm support in our examples.

File: examples/3d_iso.rs [hunk 2]
```diff
@@ -1,4 +1,6 @@
-use bevy::prelude::*;
+use std::time::Duration;
```
[the remark this answers is not in the document]
Comment (mwbryant) [reply examples/3d_iso.rs]: Done!

File: src/render/shaders/tilemap_vertex.wgsl (hunk 1 above)
Comment (mwbryant) [reply src/render/shaders/tilemap_vertex.wgsl:1]: I'm unclear what you are asking to be changed here?  Does this import not already include tilemap_data or are you saying it should not?

**Comment by rparrett** [comment]
With regard to the duplicated stuff after the shader import changes, I think something like this works.

https://gist.github.com/rparrett/b122349a0ed2a13dd3d303cd085c50db

File: src/render/shaders/tilemap_vertex.wgsl (hunk 1 above)
Comment (rparrett) [reply src/render/shaders/tilemap_vertex.wgsl:1]: My bad, I meant to place that comment in `tilemap_fragment.wgsl`.

File: src/render/shaders/tilemap_fragment.wgsl [hunk 3]
```diff
@@ -1,8 +1,11 @@
-#import bevy_ecs_tilemap::common
```
Comment (rparrett) [inline src/render/shaders/tilemap_fragment.wgsl:1]: We need `tilemap_data` too for the `atlas` feature.

Comment (mwbryant) [reply src/render/shaders/tilemap_fragment.wgsl:1]: Ah good catch, done!

**Comment by mwbryant** [comment]
That does work and makes the final result better than the original code

**Comment by rparrett** [comment]
Could you check out the shader errors in `column_even_hex.wgsl` and `column_odd_hex.wgsl`?

I'm seeing them in the `hexagon_column` and `hex_neighbors` examples. (Press space to toggle modes, some are broken)

**Comment by mwbryant** [comment]
Those examples should work now and I did a spot check over all the shaders and they all have any imports I can see.  Is there a good way to force a verification or something?  These runtime shader issues are fragile.

Also I keep moving const declaration into the functions that use them because of this shader error:
``` Composable module identifiers must not require substitution according to naga writeback rules: `bevy_ecs_tilemap::column_even_hex::SQRT_3` ```

I have no idea what this could mean or what the right solution to that is but moving the constants solves the error.

**Review by rparrett — APPROVED** [review]
> I have no idea what this could mean or what the right solution to that is but moving the constants solves the error.

I'm also a bit confused about this and looking at `naga_oil`'s source didn't really clear that up for me.

The workaround seems fine IMHO.

File: examples/accessing_tiles.rs [hunk 4]
```diff
@@ -160,9 +160,9 @@ fn main() {
...
                 .set(ImagePlugin::default_nearest()),
         )
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(update_map)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, update_map)
```
Comment (BernardIgiri) [inline examples/accessing_tiles.rs:166]: These calls can be combined. The `add_systems` method allows you to add multiple systems in one call.

File: examples/animation.rs [hunk 5]
```diff
@@ -139,12 +139,12 @@ fn main() {
...
-        .add_plugin(LogDiagnosticsPlugin::default())
-        .add_plugin(FrameTimeDiagnosticsPlugin::default())
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_startup_system(create_background)
-        .add_startup_system(create_animated_flowers)
-        .add_system(helpers::camera::movement)
+        .add_plugins(LogDiagnosticsPlugin::default())
+        .add_plugins(FrameTimeDiagnosticsPlugin::default())
+        .add_plugins(TilemapPlugin)
```
Comment (BernardIgiri) [inline examples/animation.rs:144]: These calls to `add_plugins` can be combined as the new method allows you to pass in multiple plugins at once.

File: examples/animation.rs [hunk 6]
```diff
@@ -139,12 +139,12 @@ fn main() {
...
+        .add_systems(Startup, startup)
+        .add_systems(Startup, create_background)
+        .add_systems(Startup, create_animated_flowers)
```
Comment (BernardIgiri) [inline examples/animation.rs:147]: These `add_systems` calls can be combined.

File: examples/basic.rs [hunk 7]
```diff
@@ -109,9 +109,9 @@ fn main() {
...
             ..default()
         }).set(ImagePlugin::default_nearest()))
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(swap_texture_or_hide)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, swap_texture_or_hide)
```
Comment (BernardIgiri) [inline examples/basic.rs:115]: These `add_systems` calls can  be combined.

File: examples/bench.rs [hunk 8]
```diff
@@ -56,10 +56,10 @@ fn main() {
...
             ..Default::default()
         })
-        .add_plugin(LogDiagnosticsPlugin::default())
-        .add_plugin(FrameTimeDiagnosticsPlugin::default())
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
+        .add_plugins(LogDiagnosticsPlugin::default())
+        .add_plugins(FrameTimeDiagnosticsPlugin::default())
+        .add_plugins(TilemapPlugin)
```
Comment (BernardIgiri) [inline examples/bench.rs:61]: These `add_plugins` calls can  be combined.

File: examples/chunking.rs [hunk 9]
```diff
@@ -122,11 +122,11 @@ fn main() {
...
+        .add_plugins(TilemapPlugin)
         .insert_resource(ChunkManager::default())
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(spawn_chunks_around_camera)
-        .add_system(despawn_outofrange_chunks)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, spawn_chunks_around_camera)
+        .add_systems(Update, despawn_outofrange_chunks)
```
Comment (BernardIgiri) [inline examples/chunking.rs:130]: These `add_systems` calls can  be combined.

File: examples/game_of_life.rs [hunk 10]
```diff
@@ -106,9 +106,9 @@ fn main() {
...
                 .set(ImagePlugin::default_nearest()),
         )
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(update)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, update)
```
Comment (BernardIgiri) [inline examples/game_of_life.rs:112]: These `add_systems` calls can  be combined.

File: examples/hexagon_column.rs [hunk 11]
```diff
@@ -128,9 +128,9 @@ fn main() {
...
                 .set(ImagePlugin::default_nearest()),
         )
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(swap_mesh_type)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, swap_mesh_type)
```
Comment (BernardIgiri) [inline examples/hexagon_column.rs:134]: These `add_systems` calls can  be combined.

File: examples/hexagon_row.rs [hunk 12]
```diff
@@ -127,9 +127,9 @@ fn main() {
...
                 .set(ImagePlugin::default_nearest()),
         )
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(swap_mesh_type)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, swap_mesh_type)
```
Comment (BernardIgiri) [inline examples/hexagon_row.rs:133]: These `add_systems` calls can  be combined.

File: examples/move_tile.rs [hunk 13]
```diff
@@ -67,9 +67,9 @@ fn main() {
...
                 .set(ImagePlugin::default_nearest()),
         )
-        .add_plugin(TilemapPlugin)
-        .add_startup_system(startup)
-        .add_system(helpers::camera::movement)
-        .add_system(swap_pos)
+        .add_plugins(TilemapPlugin)
+        .add_systems(Startup, startup)
+        .add_systems(Update, helpers::camera::movement)
+        .add_systems(Update, swap_pos)
```
Comment (BernardIgiri) [inline examples/move_tile.rs:73]: These `add_systems` calls can  be combined.

File: examples/hex_neighbors_radius_chunks.rs [hunk 14]
```diff
@@ -541,18 +541,18 @@ fn main() {
...
-        .add_system(hover_highlight_tile_label.after(swap_map_type))
-        .add_system(update_radius.after(hover_highlight_tile_label))
-        .add_system(highlight_neighbor_labels.after(update_radius))
+        .add_systems(Startup, (spawn_chunks, apply_deferred).chain().in_set(SpawnChunksSet))
+        .add_systems(Startup, spawn_tile_labels.after(SpawnChunksSet))
+        .add_systems(First, (camera_movement, update_cursor_pos).chain())
+        .add_systems(Update, swap_map_type)
+        .add_systems(Update, hover_highlight_tile_label.after(swap_map_type))
+        .add_systems(Update, update_radius.after(hover_highlight_tile_label))
+        .add_systems(Update, highlight_neighbor_labels.after(update_radius))
```
Comment (BernardIgiri) [inline examples/hex_neighbors_radius_chunks.rs:556]: I haven't tested this, but I believe that these `add_systems` calls can be combined as well. You should be able to combine the Startup systems in one call, and the Update systems in another call I believe.

**Review by BernardIgiri — COMMENTED** [review]
I see some opportunity to combine some of these calls to `add_systems` and `add_plugins`. Might as well take advantage of the new API functionality now.

**Review by StarArawn — APPROVED** [review]

File: examples/hex_neighbors_radius_chunks.rs (hunk 14 above)
Comment (StarArawn) [reply examples/hex_neighbors_radius_chunks.rs:556]: A great idea! I'm not going to hold the PR up for it though. :) Feel free to open another PR and I'll merge it for sure. 👍

**Pull request merged**